cli/src/cli_util.rs

```diff
@@ -1814,10 +1814,32 @@ to the current parents may contain changes from multiple commits.
                 self.write_commit_summary(formatter, &commit)?;
                 Ok(())
             });
+            error.add_hint("Immutable commits are used to protect shared history.");
+            error.add_hint(
+                "See https://jj-vcs.github.io/jj/latest/config/#set-of-immutable-commits for more \
+                 information.",
+            );
             error.add_hint(
                 "Pass `--ignore-immutable` or configure the set of immutable commits via \
                  `revset-aliases.immutable_heads()`.",
             );
+
+            let mut expression = self.attach_revset_evaluator(self.env.immutable_expression());
+            expression.intersect_with(&RevsetExpression::commits(vec![commit_id]).descendants());
+            if let Ok(to_visit_revset) = expression.evaluate() {
+                let max_evaluate = 10_000;
+                let num_descendants = to_visit_revset.iter().take(max_evaluate).count();
+                let or_more = if num_descendants >= max_evaluate {
+                    "+"
+                } else {
+                    ""
+                };
+                error.add_hint(format!(
+                    "Using `--ignore-immutable` would rewrite {num_descendants}{or_more} \
+                     immutable commit(s)."
```

Comment:
The `(s)` part is only ever relevant if it would rewrite a single commit? So I'd either check for that and print commit/commits, or (optionally) dont even print the hint for small enough number, maybe 1, maybe also configurable.

Reply:
We often simply use plural form even if the number might be 1. I personally don't care